js/data.js

```diff
@@ -34,3 +35,75 @@ function getPokemonByType(type) {
   });
   xhr.send();
 }
+
+function getPokemonSpeciesInfo(id) {
+  const description = document.querySelector('textarea');
+  const xhr = new XMLHttpRequest();
+  xhr.open('GET', 'https://pokeapi.co/api/v2/pokemon-species/' + id);
+  xhr.responseType = 'json';
+  xhr.addEventListener('load', function () {
+    xhr.response['flavor_text_entries'].forEach((entry, index) => {
+      if (entry.language.name === 'en') {
+        let x = xhr.response['flavor_text_entries'][index]['flavor_text'];
+        x = x.replaceAll('\n', ' ');
+        x = x.replaceAll('\u000c', ' ');
+        description.textContent = x;
+      }
+    });
+  });
+  xhr.send();
+}
+
+function convertHeightAndWeight(initialHeight, initialWeight) {
+  height.textContent = (initialHeight * 3.937008).toFixed(2);
+  if (height.textContent > 12) {
+    height.textContent = (height.textContent / 12).toFixed(2) + 'ft';
+  } else {
+    height.textContent = height.textContent + 'in';
+  }
+  weight.textContent = (initialWeight * 3.527396).toFixed(2);
+  if (weight.textContent > 16) {
+    weight.textContent = (weight.textContent / 16).toFixed(2) + 'lbs';
+  } else {
+    weight.textContent = weight.textContent + 'oz';
+  }
+}
+
+function getPokemonData(url) {
+  const xhr = new XMLHttpRequest();
+  xhr.open('GET', url.dataset.url);
+  xhr.responseType = 'json';
+  xhr.addEventListener('load', function () {
+    sprite.src = xhr.response.sprites.front_default;
+    if (xhr.response.sprites.back_default) {
+      backSprite.src = xhr.response.sprites.back_default;
+    } else {
+      backSprite.src = '';
+    }
+    const existingImages = document.querySelectorAll('.pokemon-type');
+    if (document.querySelectorAll) {
```

Comment:
Usually you need to pass a selector argument  to the `querySelectorAll` otherwise it returns a NodeList containing all the elements within the document ([docs](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelectorAll)). Did you mean to use `existingImages` here instead?